### DISCUSSION TITLE
Documentation and expected behaviour for evaluation groups

### DISCUSSION THREAD

**Discussion by greatvovan** [grafana/grafana#55656] [Alerting]
Reading some of the [alerting documentation](https://github.com/grafana/grafana/blob/main/docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md), I realize that it is a bit imperfect to say the least. I could probably submit an improvement PR, but first of all I would like to understand if the way the subsystem currently works is by design, and also would like to discuss about some UI decisions.

The main topic of the issue will be about evaluation groups. Also, need to note, that I am using Alerting in connection with Azure Data Explorer data source, but this part seems to be common for all types of data sources, so I believe this is relevant for all Grafana managed alerts. Please also note that documentation about [alert groupings](https://github.com/grafana/grafana/tree/main/docs/sources/alerting/alert-groups) seems to describe a different feature and out of scope of the issue.

So the documentation [says](https://github.com/grafana/grafana/blob/f8d69415ca9a515a6c4d9b8d389afea7c823d8d7/docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md?plain=1#L42):

> For Group, specify a pre-defined group. Newly created rules are appended to the end of the group.

This is not true, at least I do not see any predefined group list in my UI. Instead I am given an option to enter any string value:

<img width="333" alt="image" src="https://user-images.githubusercontent.com/4903007/191859815-e1ca526c-ca43-43d4-b5d2-a090bc73499f.png">

Another issue:

> Rules within a group are run sequentially at a regular interval, with the same evaluation time.

While this is true, it seems not complete truth. I did not look how thing works from the code perspective, but using admin tools I can observe what requests are coming to my database and when.

1. Folder name seems to contribute into determining evaluation groups. E.g. if I have alert A1 in folder F1 and alert A2 in folder F2, both having group G1, they are evaluated independently, meaning groups are defined as F1/G1 and F2/G1. I did not find this fact mentioned in the documentation. Moreover, it makes word "group" even more ambiguous as an evaluation group is not what called "Group" in the UI, but "folder/group".
2. Changing evaluation interval in a single rule results in updating an unknown number of other rules. Indeed, there is a note saying _"Evaluation interval applies to every rule within a group. It **can overwrite** the interval of an existing alert rule."_, however it is hard to understand what exactly it changes, and why word "can" is used. Can update, but may not update in some cases? When exactly?

My request to the maintainers and community:
* Please validate that item 1 above is true and by design. It was concluded from my experiments and I did not find an official confirmation to that. Although looking how Front End interacts with API, I find more confirmation to that, because in the output, rules are organized as `.data.groups[]/<group>/rules[]`, where `name`/`file` pairs are distinct, albeit it is not reflected in the UI in ay way.
* The issue 2 is a serious design flaw, in my opinion. Instead of creating/updating implicit groups, I suggest to make groups explicitly visible in the UI, for example as a "subfolder" of a folder, with UI controls (e.g. evaluation interval) visually attributable to the group.

As an outcome of this discussion, I expect either to improve documentation, or improve the UI, or both. Ideally the Front End team would implement visually intuitive user interface which leaves no doubts on how evaluation groups are organized.

**Environment**:
- Grafana version: Cloud
- Data source type & version: Azure Data Explorer
- User OS & Browser: Mac & Chrome
- Grafana plugins: Azure Data Explorer data sources v. 4.1.5

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Reply by greatvovan** [reply]
This was not a feature request.

**Reply by gelicia** [reply]
@greatvovan my apologies, most of the discussions were feature requests and the process was largely automated. Bugs/UX enhancement requests/documentation improvement requests all go in the same issue list as feature requests. If you still are encountering the same inconsistencies, feel free to convert this to an issue, and label it `type/docs` `area/alerting` so it's easier for folks to find. Thank you!
